### DISCUSSION TITLE
EtherDFS fstab mount fails when backed by nfs mount point, works at runtime

### DISCUSSION THREAD

**Discussion by jamescadd** [retronas/discussions#66] [Q&A] [answered]
Thanks for the excellent new wiki entry on FAT! I am running RetroNAS as a VM on ProxMox and have the data directory backed by a TrueNAS mount point in fstab which generally works great.

I tried to follow the steps to create a FAT loopback file that were recently added to the wiki. That works fine when I execute the commands from a prompt and the df command confirms they've succeeded when I run the mount myself from an ssh session. I created the vhd folder and file in /data/RetroNAS/vhd and mounted it to /data/RetroNAS/dos which are both under /data/RetroNAS; the mount point for TrueNAS.

When I putting the mount in fstab, the VM fails into rescue mode if I use the /data/RetroNAS/ syntax. If I instead use the syntax with IP address of the TrueNAS server that I use to also create the mount point for the data directory, the df command shows it hasn't succeeded and /data/RetroNAS/dos is still the TrueNAS exFAT filesystem (I think it said exFAT, but either way it was not the same result as running the commands manually where it said vfat or msdos with the reduced capacity of the vhd).

Any idea what I can do to troubleshoot?

**Comment by jamescadd** [comment]
@sairuk sorry to bug you, but you've helped me before with this unsupported TrueNAS setup. If you have any tips on things I may look into here they are appreciated.

**Reply by sairuk** [reply]
sorry @jamescadd i wasn't available , i see dan has answered you question

**Comment by danmons** [comment]
Couple you please copy/paste the exact contents of your RetroNAS `/etc/fstab` in here? 

"Failing into rescue mode" is normal Debian behaviour when an entry in `/etc/fstab` is wrong.  

Normally you want to test these with `sudo mount -a` on the command line which simulates what the system does at boot, iterating all of the `/etc/fstab` entries and mounting them all.  Doing that before you reboot is a good sanity test to make sure `/etc/fstab` has the correct syntax and all mounts are mountable.  Any errors from that command should be fixed before rebooting.

**Comment by jamescadd** [comment, marked as answer]
Thanks Dan, the `sudo mount -a` approach is a great tip, I'll submit a quick PR to add that to the FAT docs.  For some reason I'm now working.  In case anyone stops by later wanting to mount a FAT loopback file located on an NFS storage drive, here's what worked in my `/etc/fstab`:

10.0.0.101:/mnt/tank/retronas /storage/retronas nfs defaults
/storage/retronas/vhd/etherdfs.img      /storage/retronas/dos   msdos   loop,noatime,uid=homeuser,gid=homeuser  0 0
